src/openms/include/OpenMS/FILTERING/CALIBRATION/TrafoModel.h

```diff
@@ -0,0 +1,315 @@
+//    without specific prior written permission.
+// For a full list of authors, refer to the file AUTHORS.
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Chris Bielow $
+// $Authors: Chris Bielow $
+// --------------------------------------------------------------------------
+
+
+#ifndef OPENMS_FILTERING_CALIBRATION_TRAFOMODEL_H
+#define OPENMS_FILTERING_CALIBRATION_TRAFOMODEL_H
+
+#include <OpenMS/CONCEPT/LogStream.h>
+#include <OpenMS/DATASTRUCTURES/CalibrationData.h>
+#include <OpenMS/MATH/STATISTICS/LinearRegression.h>
+#include <OpenMS/MATH/STATISTICS/QuadraticRegression.h>
+#include <OpenMS/MATH/STATISTICS/StatisticFunctions.h>
+#include <OpenMS/MATH/MISC/RANSAC.h>
+#include <OpenMS/MATH/MISC/RANSACModelLinear.h>
+#include <OpenMS/MATH/MISC/RANSACModelQuadratic.h>
+#include <OpenMS/MATH/MISC/MathFunctions.h>
+
+#include <iterator>
+#include <map>
+#include <vector>
+
+namespace OpenMS
+{
+    
+  /**
+
+    @brief Create and apply models of a mass recalibration function.
+
+    The input is a list of calibration points (ideally spanning a wide m/z range to prevent extrapolation when applying to model).
+    
+    Models (LINEAR, LINEAR_WEIGHTED, QUADRATIC, QUADRATIC_WEIGHTED) can be trained using CalData points (or a subset of them).
+    Calibration points can have different retention time points, and a model should be build such that it captures
+    the local (in time) decalibration of the instrument, i.e. choose appropriate time windows along RT to calibrate the
+    spectra in this RT region.
+    From the available calibrant data, a model is build. Later, any uncalibrated m/z value can be fed to the model, to obtain
+    a calibrated m/z.
+
+    The input domain can either be absolute mass differences in [Th], or relative differences in [ppm].
+    The models are build based on this input.
+
+    Outlier detection before model building via the RANSAC algorithm is supported for LINEAR and QUADRATIC models.
+
+  */
+  class OPENMS_DLLAPI TrafoModel
```

Comment:
hmm we already have a TransfomationModel class so I would prefer a name like "CalibrationTransformationModel"